tests/pos/quotedPatterns.scala

```diff
@@ -14,22 +14,22 @@ object Test {
     case '{ 1 + ($y: Int)} => y
     case '{ val a = 1 + ($y: Int); 3 } => y
     case '{ val $y: Int = $z; println(`$y`); 1 } =>
-      val a: quoted.matching.Sym[Int] = y
+      val a: String = y
```

Comment:
If `y` is String, the semantics for the reference to `y` in `println($y)` becomes problematic.

Reply:
Why? Could you elaborate

Reply:
After discussion, now I see the `$y` in `val $y = e` has double meaning.

Reply:
In general, a `$` can be in an expression position (`Expr[T]`), type position (`Type[T]`) or definition position (`String`)